docs/metrics/gpt4-prompt-eng.md

```diff
@@ -0,0 +1,89 @@
+# GPT4 Prompt Engineering
+
+Prompting Engineering is a technique that involves building and optimizing specific input prompts to allow our Large Language Model (LLM) to perform better at targeted tasks. In the landscape of LLM evaluations, prompt engineering is designed to envoke the LLMs inherently knowledge and ability to reason to determine whether or not a set of inputs contain some sort of hallucination.
+
+## LLM Configurations
+
+Before diving into the techniques used for prompting we first need to understand some of the common terminology used as paramaters for LLMs.
+
+**Temperature** - Determines how the randomness of the model. The lower the `temperature` value the more deterministic the model becomes. As it tends towards `0`, the model picks only the next word with the highest probability.
+
+**Top k** - The model picks the next token from the list of top `k` tokens.
+
+**Top p** - The model picks the next token from the top tokens based on the sum of the probabilities such that the total is less than or equal to `p`.
+
+!!! warn "Limitations"
+
+    These metrics have limitations and should therefore be used at your own discretion.
+
+#  Prompting GPT4 for Hallucinations
+[GPT4](https://openai.com/gpt-4) is a closed LLM by OpenAI that we prompt in order to extract hallucinations.
+The initial specialized prompt passed into GPT4 includes the following:
+
+1. Asking GPT4 to determine in a binary fashion (strictly output yes or no) whether the generated answer contained hallucinations.
+2. Asking GPT4 for an explanation of why it answered either option.
```

Comment:
I believe the explanation was only when GPT-4 detected hallucination. cc @marko-polo-cheno